lib/plugins/helper/paginator.js

```diff
@@ -1,96 +1,136 @@
+  const {
+    current,
+    total,
+    space,
+    end_size: endSize,
+    mid_size: midSize
+  } = options;
+
+  const leftEnd = current <= endSize ? current - 1 : endSize;
+  const rightEnd = total - current <= endSize ? current + 1 : total - endSize + 1;
+  const leftMid = current - midSize <= endSize ? leftEnd + 1 : current - midSize;
+  const rightMid = current + midSize + endSize > total ? rightEnd - 1 : current + midSize;
+  const spaceHtml = htmlTag('span', { class: 'space' }, space);
+
+  const pageTag = createPageTag(options, ctx);
+
+  // Display pages on the left edge
+  for (let i = 1; i <= leftEnd; i++) {
+    tags.push(pageTag(i));
   }
 
-  if (options.show_all) {
-    // Display pages on the left side of the current page
-    for (let i = 1; i < current; i++) {
-      result += pageLink(i);
+  // Display spaces between edges and middle pages
+  if (space && current - endSize - midSize > 1) {
+    tags.push(spaceHtml);
+  }
+
+  // Display left middle pages
+  if (leftMid > leftEnd) {
+    for (let i = leftMid; i < current; i++) {
+      tags.push(pageTag(i));
     }
+  }
 
-    // Display the current page
-    result += currentPage;
+  // Display the current page
+  tags.push(pageTag(current));
 
-    // Display pages on the right side of the current page
-    for (let i = current + 1; i <= total; i++) {
-      result += pageLink(i);
-    }
-  } else {
-    // It's too complicated. May need refactor.
-    const leftEnd = current <= endSize ? current - 1 : endSize;
-    const rightEnd = total - current <= endSize ? current + 1 : total - endSize + 1;
-    const leftMid = current - midSize <= endSize ? leftEnd + 1 : current - midSize;
-    const rightMid = current + midSize + endSize > total ? rightEnd - 1 : current + midSize;
-    const spaceHtml = `<span class="space">${space}</span>`;
-
-    // Display pages on the left edge
-    for (let i = 1; i <= leftEnd; i++) {
-      result += pageLink(i);
+  // Display right middle pages
+  if (rightMid < rightEnd) {
+    for (let i = current + 1; i <= rightMid; i++) {
+      tags.push(pageTag(i));
     }
+  }
 
-    // Display spaces between edges and middle pages
-    if (space && current - endSize - midSize > 1) {
-      result += spaceHtml;
-    }
+  // Display spaces between edges and middle pages
+  if (space && total - endSize - midSize > current) {
+    tags.push(spaceHtml);
+  }
 
-    // Display left middle pages
-    if (leftMid > leftEnd) {
-      for (let i = leftMid; i < current; i++) {
-        result += pageLink(i);
-      }
-    }
+  // Dispaly pages on the right edge
+  for (let i = rightEnd; i <= total; i++) {
+    tags.push(pageTag(i));
+  }
+};
 
-    // Display the current page
-    result += currentPage;
+function paginatorHelper(options = {}) {
+  options = Object.assign({
+    base: this.page.base || '',
+    current: this.page.current || 0,
+    format: `${this.config.pagination_dir}/%d/`,
+    total: this.page.total || 1,
```

Comment:
Will `this.page.total` be zero?

Reply:
Since the initial initial value was 1, it is being handed over.